### PR TITLE
feat: fix logs for 1709

### PR DESCRIPTION
**Pull request by MarioAslau**
## **Description**

Issue: https://github.com/MetaMask/mobile-planning/issues/1709

## **Related issues**

Fixes:

## **Manual testing steps**

1. Go to this page...
2.
3.

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: app/core/EngineService/EngineService.ts [hunk 1]
```diff
@@ -123,7 +123,7 @@ class EngineService {
     controllers.forEach((controller) => {
       const { name, key = undefined } = controller;
       const update_bg_state_cb = () => {
-        if (engine.context.KeyringController.vault) {
+        if (!engine.context.KeyringController.vault) {
           Logger.message('keyringController missing for UPDATE_BG_STATE_KEY');
```
Comment (tommasini) [inline app/core/EngineService/EngineService.ts:127]: Should we change the message for 
```suggestion
          Logger.message('keyringController vault missing for UPDATE_BG_STATE_KEY');
```
?

File: app/core/EngineService/EngineService.ts [hunk 2]
```diff
@@ -111,7 +111,7 @@ class EngineService {
     }
 
     engine?.datamodel?.subscribe?.(() => {
-      if (engine.context.KeyringController.vault) {
+      if (!engine.context.KeyringController.vault) {
         Logger.message('keyringController missing for INIT_BG_STATE_KEY');
```
Comment (tommasini) [inline app/core/EngineService/EngineService.ts:115]:
```suggestion
        Logger.message('keyringController vault missing for INIT_BG_STATE_KEY');
```

**Review by tommasini — DISMISSED** [review]
LGTM will defer to you changing the message or not

**Comment by jpcloureiro** [comment]
I agree with @tommasini proposed changes on the log message.

Other than that, LGTM

**Review by jpcloureiro — DISMISSED** [review]

**Review by jpcloureiro — APPROVED** [review]

**Pull request merged**